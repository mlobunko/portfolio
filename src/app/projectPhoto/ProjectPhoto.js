import React from "react";
import PropTypes from "prop-types";

import LinkToGitHub from "shared/linkToGitHub";

import {
  StyledDescription,
  StyledNameOfProject,
  StyledProjectPhoto,
  StyledText
} from "./styles";

export const ProjectPhoto = ({
  description,
  image,
  link,
  nameOfProject,
  margin,
  textColor,
  textUp
}) => (
  <StyledProjectPhoto
    image={image}
    margin={margin}
    textColor={textColor}
    textUp={textUp}
  >
    <StyledText textUp={textUp}>
      <StyledNameOfProject>{nameOfProject}</StyledNameOfProject>
      <StyledDescription>{description}</StyledDescription>
      <LinkToGitHub
        link={link}
        nameOfProject={nameOfProject}
        textColor={textColor}
      />
    </StyledText>
  </StyledProjectPhoto>
);
ProjectPhoto.displayName = "ProjectPhoto";

export default ProjectPhoto;

ProjectPhoto.propTypes = {
  description: PropTypes.string,
  image: PropTypes.string,
  link: PropTypes.string,
  nameOfProject: PropTypes.string,
  margin: PropTypes.bool,
  textColor: PropTypes.string,
  textUp: PropTypes.bool
};
