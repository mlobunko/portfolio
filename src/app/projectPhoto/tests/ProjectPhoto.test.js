import React from "react";
import { shallow } from "enzyme";

import { ProjectPhoto } from "../ProjectPhoto";

describe("render", () => {
  it("renders correctly", () => {
    const props = {
      description: "description-family",
      image: "image",
      link: "https://a.com",
      nameOfProject: "nameOfProject",
      margin: true,
      textColor: "black",
      textUp: true
    };
    const wrapper = shallow(<ProjectPhoto {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
