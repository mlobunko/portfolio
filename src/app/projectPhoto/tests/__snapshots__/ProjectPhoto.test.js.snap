// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<StyledProjectPhoto
  image="image"
  margin={true}
  textColor="black"
  textUp={true}
>
  <StyledText
    textUp={true}
  >
    <StyledNameOfProject>
      nameOfProject
    </StyledNameOfProject>
    <StyledDescription>
      description-family
    </StyledDescription>
    <LinkToGithub
      link="https://a.com"
      nameOfProject="nameOfProject"
      textColor="black"
    />
  </StyledText>
</StyledProjectPhoto>
`;
